### PR TITLE
Update to frontend 0.0.32

### PR DESCRIPTION
**Pull request by nickcolley**
This PR updates the Design System to use the soon to be released 0.0.32 changes.

It includes guidance updates to the spacing overrides (paired with Amy), and updates to the colours page (paired with Joe)

- https://deploy-preview-325--govuk-design-system-preview.netlify.com/design-system/styles/spacing/

- https://deploy-preview-325--govuk-design-system-preview.netlify.com/design-system/styles/colour/

### CODE REVIEW

File: src/design-system/index.njk [hunk 1]
```diff
@@ -7,19 +7,19 @@ title: About
   <div class="govuk-main-wrapper govuk-main-wrapper--l">
     <div class="govuk-grid-row">
       <div class="govuk-grid-column-one-third govuk-!-margin-bottom-8">
-        <h2 class="govuk-heading-l govuk-!-w-bold">Styles</h2>
+        <h2 class="govuk-heading-l govuk-!-font-weight-bold">Styles</h2>
```
Comment (36degrees) [inline src/design-system/index.njk]: Why is this override needed when headings are bold by default?

Comment (36degrees) [reply src/design-system/index.njk]: (It's not on the other two headings in this section)

File: src/design-system/styles/layout/grid-annotate.scss [hunk 2]
```diff
@@ -6,7 +6,7 @@
 [class^="govuk-grid-column"] > p {
   @extend .govuk-body;
   display: block;
-  @include govuk-responsive-padding($govuk-spacing-responsive-6);
+  @include @include govuk-responsive-padding(6);
```
Comment (36degrees) [inline src/design-system/styles/layout/grid-annotate.scss]: Double `@include`

File: src/design-system/styles/layout/grid-nested-annotate.scss [hunk 3]
```diff
@@ -25,7 +25,7 @@
     [class^="govuk-grid-column"] > p {
       @extend .govuk-body;
       display: block;
-      @include govuk-responsive-padding($govuk-spacing-responsive-6);
+      @include @include govuk-responsive-padding(6);
```
Comment (36degrees) [inline src/design-system/styles/layout/grid-nested-annotate.scss]: Double `@include`

File: src/design-system/styles/spacing/index.md.njk [hunk 4]
```diff
@@ -109,16 +109,16 @@ If you’re building your own components and want to reference the spacing scale
...
-`padding-top: $govuk-spacing-scale-6`
+`padding-top: govuk-spacing(6)`
 
 
 ### Using the responsive spacing mixin
 
 If you want to use the responsive spacing scale you can reference the mixin.
 
-For example, use `@include govuk-responsive-margin($govuk-spacing-responsive-6, "bottom");` for 30px margin-bottom on large screens and 20px on small screens.
+For example, use `@include @include govuk-responsive-margin(6, "bottom");` for 30px margin-bottom on large screens and 20px on small screens.
```
Comment (36degrees) [inline src/design-system/styles/spacing/index.md.njk]: Double `@include`

File: src/design-system/styles/spacing/index.md.njk [hunk 5]
```diff
@@ -109,16 +109,16 @@ If you’re building your own components and want to reference the spacing scale
...
 
-Use `@include govuk-responsive-padding($govuk-spacing-responsive-6);` for 30px padding on all sides on large screens and 20px on small screens.
+Use `@include @include govuk-responsive-padding(6);` for 30px padding on all sides on large screens and 20px on small screens.
```
Comment (36degrees) [inline src/design-system/styles/spacing/index.md.njk]: Double `@include`

File: src/design-system/stylesheets/components/_contact-panel.scss [hunk 6]
```diff
@@ -1,8 +1,8 @@
 
 .app-contact-panel {
-  @include govuk-responsive-margin($govuk-spacing-responsive-6, "top");
-  @include govuk-responsive-margin($govuk-spacing-responsive-6, "bottom");
-  @include govuk-responsive-padding($govuk-spacing-responsive-3);
+  @include @include govuk-responsive-margin(6, "top");
```
Comment (36degrees) [inline src/design-system/stylesheets/components/_contact-panel.scss]: More double `@include`s

Comment (36degrees) [reply src/design-system/stylesheets/components/_contact-panel.scss]: There's more in this commit but not going to comment on them all…

File: src/design-system/stylesheets/components/_contact-panel.scss [hunk 7]
```diff
@@ -25,6 +25,6 @@
 }
 
 .app-contact-panel__body {
-  @include govuk-font-regular-19
+  @include govuk-font(19, $weight: regular)
```
Comment (36degrees) [inline src/design-system/stylesheets/components/_contact-panel.scss]: Regular is the default value for `$weight` so can be omitted wherever you're calling `govuk-font`.

File: src/design-system/stylesheets/components/_cookie-banner.scss [hunk 8]
```diff
@@ -1,15 +1,15 @@
 .app-cookie-banner {
-  @include govuk-font-regular-16;
+  @include govuk-font(16, $weight: regular);
```
Comment (igloosi) [inline src/design-system/stylesheets/components/_cookie-banner.scss]: no need for regular

File: src/design-system/stylesheets/components/_cookie-banner.scss [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
-  padding-top: $govuk-spacing-scale-3;
-  padding-right: $govuk-spacing-scale-3;
-  padding-bottom: $govuk-spacing-scale-3;
-  padding-left: $govuk-spacing-scale-3;
-  background-color: $govuk-light-blue-25;
+  padding-top: govuk-spacing(3);
+  padding-right: govuk-spacing(3);
+  padding-bottom: govuk-spacing(3);
+  padding-left: govuk-spacing(3);
+  background-color: govuk-colour("light-blue")-25;
```
Comment (igloosi) [inline src/design-system/stylesheets/components/_cookie-banner.scss]: extra -25?

File: src/design-system/stylesheets/components/_example.scss [hunk 10]
```diff
@@ -1,20 +1,20 @@
 // Example styles
 .app-example-wrapper {
-  @include govuk-responsive-margin($govuk-spacing-responsive-6, "top");
-  @include govuk-responsive-margin($govuk-spacing-responsive-6, "bottom");
-  @include govuk-font-regular-19;
+  @include govuk-responsive-margin(6, "top");
+  @include govuk-responsive-margin(6, "bottom");
+  @include govuk-font(19, $weight: regular);
```
Comment (igloosi) [inline src/design-system/stylesheets/components/_example.scss]: no need for regular

File: src/design-system/stylesheets/components/_cookie-banner.scss (hunk 9 above)
Comment (36degrees) [reply src/design-system/stylesheets/components/_cookie-banner.scss]: Somewhere we were actually using the tints 😱

**Comment by igloosi** [comment]
not going to mention every instance of an extra 'regular' param in the font mixin

File: src/design-system/stylesheets/components/_cookie-banner.scss (hunk 9 above)
Comment (igloosi) [reply src/design-system/stylesheets/components/_cookie-banner.scss]: quick, let's get dave to suggest a colour :)

Comment (36degrees) [reply src/design-system/stylesheets/components/_cookie-banner.scss]: I think lighten is relative, I think you'll need to use `scale-color` or `mix-color` and mix it with white. Colour manipulation isn't one of my strong points… guess check the result against the colour in master!

Comment (igloosi) [reply src/design-system/stylesheets/components/_cookie-banner.scss]: lighten(desaturate(govuk-colour("light-blue"), 8.46), 42.55) :)

File: src/design-system/styles/layout/grid-annotate.scss (hunk 2 above)
Comment (nickcolley) [reply src/design-system/styles/layout/grid-annotate.scss]: fixed in later commit (that i didnt fixup yet)...

**Comment by igloosi** [comment]
looks like all has been updated and should Just work <sup>**TM**</sup> when Frontend is released

**Pull request merged**